server/map_rpc_server.go

```diff
@@ -166,6 +166,12 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 			return nil, status.Errorf(codes.InvalidArgument,
 				"len(%x): %v, want %v", l.Index, got, want)
 		}
+		if l.LeafValue == nil {
```

Comment:
I was slightly surprised that this isn't an error but I'm not certain of the map semantics here.